app/worker/http.go

```diff
@@ -39,34 +40,41 @@ var makeHTTPRequest = func(req *http.Request, timeout time.Duration) (*http.Resp
 	return res, nil
 }
 
-func processHTTPTask(httpTask *model.HTTPTask, job *model.Job,
-	workflow *model.Workflow, l *log.Logger) (*model.TaskResult, error) {
-	uri := processJobString(httpTask.URI, workflow, job)
+func processHTTPTask(
+	httpTask *model.HTTPTask,
+	ps *processor.JobStringProcessor,
+	jp *processor.JobProcessor,
+	l *log.Logger,
+) (*model.TaskResult, error) {
+	options := &processor.Options{Encoding: processor.URL}
+	uri := ps.ProcessJobString(httpTask.URI, options)
```

Comment:
This is going to break certain URLs. For example if the url contains a `host:port` address. To be honest, I think the query escape should be done at the templating level and thus done as part of the workflow definition.

Reply:
wait how can it be? it only URL-encodes the value:

[app/processor/string.go:97](https://github.com/mendersoftware/workflows/pull/234/files#diff-a395ff9022c3014278f32fad3c86b421391bc1558be6cf5ff6539e33cbbcbd89R97)
```go
						value = url.QueryEscape(value)
```

Reply:
This transforms all of the following characters: `'$', '&', '+', ',', '/', ':', ';', '=', '?', '@'` many of them are valid in different parts of the schema. For example
```go
url.QueryEscape("https://hosted.mender.io") // -> https%3A%2F%2Fhosted.mender.io
```

Reply:
but the URL is never a workflow input variable, we take it only from env and with default values (`"uri": "http://${env.USERADM_ADDR|mender-useradm:8080}/api/internal/v1/useradm/tokens?tenant_id=${workflow.input.tenant_id}"`). this assumes that we encode the workflow input variables present in the URL (`${workflow.input.tenant_id}`).

Reply:
ok, how about now? we support cool `encoding=plain|url;identifier` syntax that overrides everything you may pass. if you do not pass it, by default we URL encode all input variables that exist in the URL.

Reply:
@tranchitella I hope you are happy that you used Alf as a mole to have this feature :>